src/engine/controls/cuecontrol.cpp

```diff
@@ -347,10 +349,17 @@ void CueControl::trackLoaded(TrackPointer pNewTrack) {
     loadCuesFromTrack();
 
     // Seek track according to SeekOnLoadMode.
-    SeekOnLoadMode seekOnLoadMode = getSeekOnLoadMode();
+    SeekOnLoadMode seekOnLoadMode = getSeekOnLoadModeAutoDjOverride();
+    if (seekOnLoadMode == SeekOnLoadMode::UsePreference) {
+        seekOnLoadMode = getSeekOnLoadPreference();
+    }
     switch (seekOnLoadMode) {
-      case SeekOnLoadMode::Beginning:
-        seekExact(0.0);
+    case SeekOnLoadMode::Beginning:
+        // This allows users to load tracks and have the needle-drop be maintained.
+        if (!(m_pVinylControlEnabled->get() &&
+                m_pVinylControlMode->get() == MIXXX_VCMODE_ABSOLUTE)) {
```

Comment:
I'm not sure if this should be checked for any other SeekOnLoadModes. @ywwg maybe you can clarify?